src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -38,10 +38,6 @@ namespace Dnssd {
 
 namespace {
 
-#if CHIP_CONFIG_MDNS_CACHE_SIZE > 0
-static DnssdCache<CHIP_CONFIG_MDNS_CACHE_SIZE> sDnssdCache;
```

Comment:
Why is the caching here being removed?

Reply:
The DNS cache story is basically odd and I would say wrong:
   - we double-cache: CASESessionManager has its own DNSCache that it tries to use (but realistically this PR nixes some uses because we do not go through CASESessionManager)
   - Platform DNS servers have their own caches since they are shared among binaries. They store TTL and having an extra layer of caching in front of it is not that useful (or it does not seem so). We _would_ benefit in minmdns, but that is exactly the place where a cache is not used.
   
 I would have a followup on fixing the cache story by adding it to address resolver. It is somewhat unclear how to use it though: we need invalidation on CASE failures for example,  which we currently did not do.
 
 The PR grows a bit much already.

Reply:
After offline discussion, created #16029 to asses smoke test impact.